content/ADR-148-processing-scene-messages.md

```diff
@@ -0,0 +1,113 @@
+
+<figure>
+  <img src="/resources/ADR-148/Frame 1.svg" />
+</figure>
+
+#### Implications
+
+- The scene will run at half the speed of the renderer in the best case sceneario, introducing Dropped frames.
+- The renderer will maintain full speed in the rendering, but every other frame will be the same as the previous one.
+- If the update of the scene reaches the renderer mid-frame, then two frames will be dropped for the scene.
+
+### Extension
+
+To illustrate better what happens inside the scene frame and renderer frame, we will consider some stages for each of them:
+
+For the scene frame:
+- **Scene.Recieve**: Read all the messages from the renderer
+- **Scene.Update**: Execute scene logic, run systems, mutate state
+- **Scene.Send**: Send all the updates to the renderer
+
+For the renderer frame:
+- **Renderer.Recieve**: Read all the messages from the scenes
+- **Renderer.Update**: Update transforms, calculate physics, etc.
+- **Renderer.Render**: Batch GPU commands and send them to the GPU Process
+- **Renderer.Send**: Send all the updates to the scene
+
+<figure>
+  <img src="/resources/ADR-148/Frame 2.svg" />
+</figure>
+
+### Proposed approach
+
+Now that the frame is decomposed into smaller chunks, it can be observed that the RENDER part doesn't necessarily needs to halt the **Renderer.Send** back to the scene, but it is a requirement for the **Renderer.Render**, since physics and transformations are used to calculate the GPU buffers for the next frame.
+
+<figure>
+  <img src="/resources/ADR-148/Frame 4.svg" />
+</figure>
+
+An extension to this optimization is that the **Renderer.Receive** can happen in parallel, while the previous GPU frame is still being processed by the GPU Process. Effectively removing the dropped scene frames caused by the excessive waits.
+
+<figure>
+  <img src="/resources/ADR-148/Frame 6.svg" />
+</figure>
+
+> Considerations: This approach is way better on multi-threaded systems. Since it would be possible to parallelize the rendering and the **Renderer.Send**
+
+#### Working with multiple scenes
+
+The scenes runtimes are RECOMMENDED to be independant and to run in parallel. It is also RECOMMENDED that the Renderer can process those updates concurrently.
+
+There is one explicit synchronization point that implementers MUST consider: The renderer MUST NOT respond to the scene until all the messages of the previous frame were processed and the physics and camera position were calculated.
+
+It was considered for this design that a scene in the renderer can take several renderer frames to process all the queued messages. Implementers SHOULD process all the messages from scenes in order and MUST prioritize first "global scenes" and then scenes ordered by distance.
+
+If scenes are too far away, it MAY be possible that those will receive eventual updates because the closest scenes MAY consume most of the processing quota.
```

Comment:
```suggestion
If scenes are too far away, it MAY be possible that those will receive eventual updates because the closest sceneries MAY consume most of the processing quota.
```

Reply:
It was just a synonym to avoid repeating scenes too much. I think we can use "scenes"